scripts/main.py

```diff
@@ -345,20 +353,22 @@ def parse_arguments():
             "NEXT ACTION: ",
             Fore.CYAN,
             f"COMMAND = {Fore.CYAN}{command_name}{Style.RESET_ALL}  ARGUMENTS = {Fore.CYAN}{arguments}{Style.RESET_ALL}")
-
+    if user_input == "RETRY":
+            continue
     # Execute command
-    if command_name.lower() != "error":
-        result = f"Command {command_name} returned: {cmd.execute_command(command_name, arguments)}"
-    else:
-        result = f"Command {command_name} threw the following error: " + arguments
-
-    # Check if there's a result from the command append it to the message
-    # history
-    if result is not None:
-        full_message_history.append(chat.create_chat_message("system", result))
-        print_to_console("SYSTEM: ", Fore.YELLOW, result)
-    else:
-        full_message_history.append(
-            chat.create_chat_message(
-                "system", "Unable to execute command"))
-        print_to_console("SYSTEM: ", Fore.YELLOW, "Unable to execute command")
+    if user_input != "RETRY":
```

Comment:
This is redundant as there is a comparison above already. I suggest you to just remove the if, so that the unchanged code below is showed as unchanged in diff. Then you can submit a further improvement as a separate PR.